go/fake/fake_sdk.go

```diff
@@ -120,6 +142,31 @@ func (m *mockHandler) MockServiceCallMethod(serviceName string, methodName strin
 	m.serviceStubs = append(m.serviceStubs, serviceStub{key: key, out: out})
 }
 
+func (m *mockHandler) MockEmitEvent(eventFunctionSignature interface{}, args ...interface{}) {
+	var key []interface{}
+	//	key = append(key, eventFunctionSignature) // is there a way to equal funcs?
+	key = append(key, args...)
+	m.eventStubs = append(m.eventStubs, eventStub{key: key})
+}
+
+func (m *mockHandler) VerifyMocks() {
```

Comment:
question: should we call `VerifyMocks` after the test finished running automatically? this could be achieved in the "test framework" (the `inscope` function)